src/lib.rs

```diff
@@ -38,6 +38,18 @@ pub use derive_arbitrary::*;
 #[doc(inline)]
 pub use unstructured::Unstructured;
 
+/// Error indicating that the maximum recursion depth has been reached while calculating [`Arbitrary::size_hint`]()
+#[derive(Debug, Clone, Copy)]
+pub struct MaxRecursionReached;
```

Comment:
For future-proofing, let's make this have a private field and avoid deriving `Copy`:

```suggestion
#[derive(Debug, Clone)]
pub struct MaxRecursionReached {
    _private: ()
}
```

Reply:
Why use a `_private` field instead of `#[non_exhaustive]` ?

Reply:
Should we also implement `Default` for it ?

Reply:
> Why use a `_private` field instead of `#[non_exhaustive]` ?

No reason, I just always forget new Rust features.

Reply:
> Should we also implement `Default` for it ?

If there isn't a strong reason to implement it, then I'd prefer to avoid it. It gives us more freedom down the line if we decide to do something like add the recursion depth or type names or something to the error, in which case a default might be tricky to choose.

Reply:
Without default, the only way for users to create it is to use `size_hint::try_recursion_guard`. Maybe there's some weird use case where someone wants to return this as an error without going through this function.

Reply:
I think that's probably okay, if someone needs it then we can cross that bridge at that time.